src/win/fs.c

```diff
@@ -1456,11 +1563,8 @@ void fs__scandir(uv_fs_t* req) {
           info->FileName[1] == L'.')
         continue;
 
-      /* Compute the space required to store the filename as UTF-8. */
-      utf8_len = WideCharToMultiByte(
-          CP_UTF8, 0, &info->FileName[0], wchar_len, NULL, 0, NULL, NULL);
-      if (utf8_len == 0)
-        goto win32_error;
+      /* Compute the space required to store the filename as WTF-8. */
+      wtf8_len = fs__get_length_wide(&info->FileName[0], wchar_len);
```

Comment:
What's the worst case here, `wchar_len * 4`? I wouldn't bother with a size calculation and just allocate that.

Reply:
It is what libuv has always done for this API. I think it is worthwhile trying not to 4x most allocations, since we cannot easily shrink them, and it will rarely be needed beyond 1x.